plugin/src/main/scala/org/virtuslab/akkasaferserializer/SaferSerializerPluginComponent.scala

```diff
@@ -0,0 +1,61 @@
+package org.virtuslab.akkasaferserializer
+
+import akka.actor.typed.Behavior
+
+import scala.tools.nsc.plugins.PluginComponent
+import scala.tools.nsc.{Global, Phase}
+
+class SaferSerializerPluginComponent(val pluginOptions: PluginOptions, val global: Global) extends PluginComponent {
+  import global._
+  override val phaseName: String = "akka-safer-serializer-gather"
+  override val runsAfter: List[String] = List("refchecks")
+
+  var rootsCache: List[Type] = List()
```

Comment:
Actually... what's the lifetime of a single `SaferSerializerPluginComponent` object? Is it e.g. cached b/w separate invocations of `compile` in a single `sbt` session? Maybe it'd be a better idea to make the phase completely stateless (everything kept in local vars & method params and not fields?)

Reply:
From my experiments, it's a single compilation run. That's why it's called cache, it's completely optional. We could remove it without any repercussion, besides a longer compilation time. In most projects, there will only be one trait, and most messages will already be extending it, so performance gain may be significant.

Reply:
Ok, if so then let's leave as is